topostats/processing.py

```diff
@@ -77,10 +77,11 @@ def process_scan(
     core_out_path = get_out_path(image_path, base_dir, output_dir).parent / "processed"
     core_out_path.mkdir(parents=True, exist_ok=True)
     filter_out_path = core_out_path / filename / "filters"
```

Comment:
To fix the tests which currently fail we need to move the

```python
grain_out_path = core_out_path / filename / "grains"
```

to here, but leave all the `*.mkdir()` calls within the `if plotting_config["image_set"] == "all":`

Apologies for not clocking/understanding/testing this in my original suggestion/comment.